src/operator/src/statement/ddl.rs

```diff
@@ -518,8 +518,54 @@ impl StatementExecutor {
             expr
         );
 
-        let req = SubmitDdlTaskRequest {
-            task: DdlTask::new_alter_table(expr.clone()),
+        let physical_table_id = self
+            .table_metadata_manager
+            .table_route_manager()
+            .get_physical_table_id(table_id)
+            .await
+            .context(TableMetadataManagerSnafu)?;
+
+        let (req, invalidate_keys) = if physical_table_id == table_id {
+            // This is physical table
+            let req = SubmitDdlTaskRequest {
+                task: DdlTask::new_alter_table(expr),
+            };
+
+            let mut invalidate_keys = vec![
+                CacheIdent::TableId(physical_table_id),
+                CacheIdent::TableId(table_id),
```

Comment:
Why do we need to invalidate the table id twice? Since we already checked `physical_table_id == table_id`

Reply:
Very good catch, this code was placed in the wrong position. 😄